### ISSUE TITLE
Preboot buffer doesn't start angular2 app after rendering

### ISSUE THREAD

**Issue by jeffwhelpley** [angular/preboot#2]
Transfer of issue for @jaumard from https://github.com/angular/universal/issues/482
- **I'm submitting a ...**
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support request here, see note at the top of this template.
- **What is the current behavior?**
  When we load a page, server rendering render the page correctly (the first time only because of this bu https://github.com/angular/universal/issues/481) But once rendering is done angular 2 app doesn't seems to be started because if we click on router link the page is loaded instead of just the component.
- **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem** by creating a github repo.
  Demo is here https://github.com/jaumard/trails-angular2-isomorphic, go to `api/controllers/ViewController` and enable the buffer option, then `npm start` and go to `localhost:3000` if you click on Home or About you'll see the page is reload, router doesn't do his job.
- **What is the expected behavior?**
  Once page is rendered angular2 app should take the lead like when buffer option is false
- **Please tell us about your environment:**
- Angular version: 2.0.0-rc4
- Browser: [all]
- Language: [all]
- OS:  [Mac OS X]
- Platform: [NodeJs]

**Comment by jotto** [comment]
I _think_ I have a similar issue with Angular 1.x, but was strangely able to resolve it but putting a view inside the app container.

## fails
```html
<div ng-app ng-view></div>
```

## works
```html
<div ng-app>
  <div ng-view></div>
</div>
```
but this _experiment_ required a `preboot-inline` modification:

```javascript
// obviously we would search/detect whether
// ng-app has an ng-view child before appending this
var ngView = document.createElement('div');
ngView.setAttribute('data-ng-view', '')
rootClientNode.appendChild(ngView)
```

**Comment by jeffwhelpley** [comment]
I am working on a pretty significant refactor of preboot now to make it easier to integrate with ng v4. I know this issue is for v1.x, but I will revisit this once v4 is out.

**Comment by jotto** [comment]
an update to my previous comment:

instead of creating an `ng-view` for Angular v1.x, I changed the `preboot-inline` to do a deep clone: `cloneNode(true)` instead of a shallow clone.

**Comment by jeffwhelpley** [comment]
This should be fixed with 5.0.0. Please try out 5.0.0-rc.1 or above.
